driver/src/solution_submission.rs

```diff
@@ -103,33 +103,57 @@ impl<'a> StableXSolutionSubmitter<'a> {
             gas_price_estimating,
         }
     }
+
+    /// Turn a method error from a solution submission into a SolutionSubmissionError.
+    async fn make_error(
+        &self,
+        batch_index: U256,
+        solution: Solution,
+        err: MethodError,
+    ) -> SolutionSubmissionError {
```

Comment:
This was hard to read in the original function and very hard to convert to async inline so I refactored it into a separate function.

Reply:
:+1: I like the helper function as well